content/research/market-health/posts/2024-05-01-floki-wash-trading/analysis.py

```
@@ -0,0 +1,110 @@
+    n=len(df)
+    top1=vc.iloc[0]/n
+    top10=vc.head(10).sum()/n
+    # exact round lots: qty divisible by 100000 (round 100k+) and >0
+    q=df["qty"].values
+    round_mil=np.mean((q>0)&(np.mod(q,500000.0)==0)) if df["qty"].median()>1 else np.mean((q>0)&(np.mod(np.round(q,8),0.01)==0)&(q>=0.01))
+    return float(top1),float(top10),float(round_mil),vc
+
+for s in ["FLOKIUSDT","BTCUSDT"]:
+    df,sha,size=D[s]
+    top1,top10,rmil,vc=lot_metrics(df)
+    q=df["quote"].values; q=q[q>0]
```

Ruff (0.15.17)

[error] Multiple statements on one line (semicolon)

(E702)